common/src/main/java/com/wynntils/models/items/encoding/impl/block/CustomGearTypeTransformer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.items.encoding.impl.block;
+
+import com.wynntils.models.gear.type.GearType;
+import com.wynntils.models.items.encoding.data.CustomGearTypeData;
+import com.wynntils.models.items.encoding.type.DataTransformer;
+import com.wynntils.models.items.encoding.type.ItemTransformingVersion;
+import com.wynntils.utils.type.ArrayReader;
+import com.wynntils.utils.type.ErrorOr;
+import com.wynntils.utils.type.Pair;
+import com.wynntils.utils.type.UnsignedByte;
+import java.util.List;
+
+public class CustomGearTypeTransformer extends DataTransformer<CustomGearTypeData> {
+    private static final List<Pair<GearType, Integer>> GEAR_TYPE_IDS = List.of(
+            new Pair<>(GearType.SPEAR, 0),
```

Comment:
Maybe these ids should be stored in the GearType itself?

Reply:
I had a general rule when creating these transformers. If I added enums that are item encoding/transformer specific, I store the IDs there. However, if I use enums from other parts of the mod, like gear enums, I'll map them to an id in the transformer itself. This is mainly done to not have a lot of enum have some kind of "magic" ID. What do you think, should I add them to the enums, or keep it this way?

Reply:
I think we can add the "magic" IDs to the enum. Compare it with e.g. the API name. In a way that is a "magic" key for the Wynncraft API. And this ID is just another mapping between our data types and some other external interface. As long as we're not downing in such interfaces, I think it makes sense to store it as part of our definition.

Reply:
Okay, sure. It's done.